modules/amm-executor/src/main/scala/org/ergoplatform/dex/executor/amm/processes/Executor.scala

```diff
@@ -48,34 +51,50 @@ object Executor {
     }
 
   final private class Live[
-    F[_]: Monad: Evals[*[_], G],
-    G[_]: Monad: Logging: TraceId.Local: Clock: Catches
+    F[_]: Monad: Evals[*[_], G]: Defer: SemigroupK: ParFlatten,
+    G[_]: Monad: Logging: Catches: Timer
   ](conf: ExecutionConfig)(implicit
     orders: CFMMCircuit[F, G],
+    executedOrders: CFMMHistConsumer[F, G],
+    backlog: CFMMBacklog[G],
     service: Execution[G],
     errParser: TxSubmissionErrorParser
   ) extends Executor[F] {
 
     def run: F[Unit] =
+      emits(
+        List(
+          addToBacklog,
+          executeOrders,
+          dropExecuted
+        )
+      ).parFlattenUnbounded
+
+    def addToBacklog: F[Unit] =
       orders.stream
-        .evalMap { rec =>
-          service
-            .executeAttempt(rec.message)
-            .handleWith[Throwable](e => warnCause"Order execution failed fatally" (e) as none[CFMMOrder])
-            .local(_ => TraceId.fromString(rec.message.id.value))
-            .tupleLeft(rec)
+        .evalTap(orderRec => backlog.put(orderRec.message))
+        .evalMap(_.commit)
```

Comment:
For Kafka it's inefficient to commit messages by one

Reply:
We can't batch orders, because of it will slow down execution